ddtrace/commands/ddtrace_run.py

```diff
@@ -16,54 +27,89 @@
     else:
-        os.environ['PYTHONPATH'] = bootstrap_dir
+        os.environ["PYTHONPATH"] = bootstrap_dir
 
 
 def main():
-    if len(sys.argv) < 2 or sys.argv[1] == '-h':
-        print(USAGE)
-        return
-
-    log.debug('sys.argv: %s', sys.argv)
-
-    root_dir = _ddtrace_root()
-    log.debug('ddtrace root: %s', root_dir)
-
-    bootstrap_dir = os.path.join(root_dir, 'bootstrap')
-    log.debug('ddtrace bootstrap: %s', bootstrap_dir)
+    parser = argparse.ArgumentParser(
+        description=USAGE,
+        prog="ddtrace-run",
+        usage="ddtrace-run <your usual python command>",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument("command", nargs=argparse.REMAINDER, type=str, help="Command string to execute.")
+    parser.add_argument("-d", "--debug", help="enable debug mode (disabled by default)", action="store_true")
+    parser.add_argument("-i", "--info", help="print library info useful for debugging", action="store_true")
+    parser.add_argument("-p", "--profiling", help="enable profiling (disabled by default)", action="store_true")
+    parser.add_argument("-v", "--version", action="version", version="%(prog)s " + ddtrace.__version__)
+    args = parser.parse_args()
+
+    if args.profiling:
+        os.environ["DD_PROFILING_ENABLED"] = "true"
+
+    if args.debug:
+        os.environ["DD_TRACE_DEBUG"] = "true"
+
+    if args.info:
+        # Inline imports for performance.
+        import pprint
+        from ddtrace.internal.debug import collect
+
+        pprint.pprint(collect(ddtrace.tracer))
+        sys.exit(0)
+
+    root_dir = os.path.dirname(ddtrace.__file__)
+    log.debug("ddtrace root: %s", root_dir)
+
+    bootstrap_dir = os.path.join(root_dir, "bootstrap")
+    log.debug("ddtrace bootstrap: %s", bootstrap_dir)
 
     _add_bootstrap_to_pythonpath(bootstrap_dir)
-    log.debug('PYTHONPATH: %s', os.environ['PYTHONPATH'])
-    log.debug('sys.path: %s', sys.path)
+    log.debug("PYTHONPATH: %s", os.environ["PYTHONPATH"])
+    log.debug("sys.path: %s", sys.path)
 
-    executable = sys.argv[1]
+    if not args.command:
+        parser.print_help()
+        sys.exit(1)
 
     # Find the executable path
-    executable = spawn.find_executable(executable)
-    log.debug('program executable: %s', executable)
-
-    os.execl(executable, executable, *sys.argv[2:])
+    executable = spawn.find_executable(args.command[0])
+    if not executable:
+        print("ddtrace-run: failed to find executable '%s'.\n" % args.command[0])
+        parser.print_usage()
+        sys.exit(1)
+
+    log.debug("program executable: %s", executable)
+
+    try:
+        # Raises OSError for permissions errors in Python 2
+        #        PermissionError for Python 3
+        os.execl(executable, executable, *args.command[1:])
+    except (OSError, PermissionError):
+        print("ddtrace-run: executable '%s' does not have executable permissions.\n" % executable)
+        parser.print_usage()
```

Comment:
do we need to print usage here?

Reply:
is this the short usage info or the full help info?

Reply:
From the example in the PR description above:
![image](https://user-images.githubusercontent.com/6321485/94170845-8f226b80-fe5e-11ea-988b-2353a999723f.png)